lib/color.cpp

```diff
@@ -40,26 +42,39 @@ static bool isStreamATty(const std::ostream & os)
 }
 #endif
 
-std::ostream& operator<<(std::ostream & os, Color c)
+static bool isColorEnabled(const std::ostream & os)
 {
-#ifndef _WIN32
-    if (!gDisableColors && isStreamATty(os))
-        return os << "\033[" << static_cast<std::size_t>(c) << "m";
+    // See https://bixense.com/clicolors/
+    static const bool color_forced_off = (nullptr != std::getenv("NO_COLOR"));
+    if (color_forced_off)
+    {
+        return false;
+    }
+    static const bool color_forced_on = (nullptr != std::getenv("CLICOLOR_FORCE"));
+    if (color_forced_on)
+    {
+        return true;
+    }
+#ifdef _WIN32
+    (void)os;
+    return false;
 #else
-    (void)c;
+    return isStreamATty(os);
 #endif
+}
+
+std::ostream& operator<<(std::ostream & os, Color c)
+{
+    if (!gDisableColors && isColorEnabled(os))
+        return os << "\033[" << static_cast<std::size_t>(c) << "m";
+
     return os;
 }
 
 std::string toString(Color c)
 {
-#ifndef _WIN32
-    std::stringstream ss;
+    std::ostringstream ss;
```

Comment:
I was about to comment that I would prefer if we could do this without a stream but it turns out this function is only called to generate a static lookup table so it is actually fine.